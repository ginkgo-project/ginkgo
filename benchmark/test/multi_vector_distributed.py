#!/usr/bin/env python3
import test_framework

# check that all input modes work:
# parameter
test_framework.compare_output_distributed(
    ["-input", '[{"n": 100}]'],
    expected_stdout="multi_vector_distributed.simple.stdout",
    expected_stderr="multi_vector_distributed.simple.stderr",
    num_procs=3,
)

# stdin
test_framework.compare_output_distributed(
    [],
    expected_stdout="multi_vector_distributed.simple.stdout",
    expected_stderr="multi_vector_distributed.simple.stderr",
    stdin='[{"n": 100}]',
    num_procs=3,
)

# file
test_framework.compare_output_distributed(
    ["-input", str(test_framework.sourcepath / "input.blas.json")],
    expected_stdout="multi_vector_distributed.simple.stdout",
    expected_stderr="multi_vector_distributed.simple.stderr",
    num_procs=3,
)

# profiler annotations
test_framework.compare_output_distributed(
    ["-input", '[{"n": 100}]', "-profile", "-profiler_hook", "debug"],
    expected_stdout="multi_vector_distributed.profile.stdout",
    expected_stderr="multi_vector_distributed.profile.stderr",
    num_procs=3,
)

# complex
test_framework.compare_output_distributed(
    ["-input", '[{"n": 100}]'],
    expected_stdout="multi_vector_distributed_dcomplex.simple.stdout",
    expected_stderr="multi_vector_distributed_dcomplex.simple.stderr",
    num_procs=3,
    use_complex=True
)
