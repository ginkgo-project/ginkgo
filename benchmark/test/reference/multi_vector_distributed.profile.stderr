Running on reference(0)
Running with 0 warm iterations and 1 running iterations
The random seed for right hand sides is 42
The operations are copy,axpy,scal
Running test case n = 100 
DEBUG: begin n = 100 
	Running blas: copy
DEBUG: begin copy
DEBUG: begin partition::build_ranges_from_global_size
DEBUG: end   partition::build_ranges_from_global_size
DEBUG: begin components::fill_array
DEBUG: end   components::fill_array
DEBUG: begin components::fill_array
DEBUG: end   components::fill_array
DEBUG: begin components::fill_array
DEBUG: end   components::fill_array
DEBUG: begin components::fill_array
DEBUG: end   components::fill_array
DEBUG: begin partition::build_from_contiguous
DEBUG: end   partition::build_from_contiguous
DEBUG: begin partition::build_starting_indices
DEBUG: end   partition::build_starting_indices
DEBUG: begin copy
DEBUG: end   copy
DEBUG: begin copy
DEBUG: end   copy
DEBUG: begin partition::build_ranges_from_global_size
DEBUG: end   partition::build_ranges_from_global_size
DEBUG: begin components::fill_array
DEBUG: end   components::fill_array
DEBUG: begin components::fill_array
DEBUG: end   components::fill_array
DEBUG: begin components::fill_array
DEBUG: end   components::fill_array
DEBUG: begin components::fill_array
DEBUG: end   components::fill_array
DEBUG: begin partition::build_from_contiguous
DEBUG: end   partition::build_from_contiguous
DEBUG: begin partition::build_starting_indices
DEBUG: end   partition::build_starting_indices
DEBUG: begin copy
DEBUG: end   copy
DEBUG: begin copy
DEBUG: end   copy
DEBUG: begin dense::fill
DEBUG: end   dense::fill
DEBUG: begin repetition
DEBUG: begin dense::copy
DEBUG: end   dense::copy
DEBUG: end   repetition
DEBUG: end   copy
	Running blas: axpy
DEBUG: begin axpy
DEBUG: begin partition::build_ranges_from_global_size
DEBUG: end   partition::build_ranges_from_global_size
DEBUG: begin components::fill_array
DEBUG: end   components::fill_array
DEBUG: begin components::fill_array
DEBUG: end   components::fill_array
DEBUG: begin components::fill_array
DEBUG: end   components::fill_array
DEBUG: begin components::fill_array
DEBUG: end   components::fill_array
DEBUG: begin partition::build_from_contiguous
DEBUG: end   partition::build_from_contiguous
DEBUG: begin partition::build_starting_indices
DEBUG: end   partition::build_starting_indices
DEBUG: begin copy
DEBUG: end   copy
DEBUG: begin copy
DEBUG: end   copy
DEBUG: begin partition::build_ranges_from_global_size
DEBUG: end   partition::build_ranges_from_global_size
DEBUG: begin components::fill_array
DEBUG: end   components::fill_array
DEBUG: begin components::fill_array
DEBUG: end   components::fill_array
DEBUG: begin components::fill_array
DEBUG: end   components::fill_array
DEBUG: begin components::fill_array
DEBUG: end   components::fill_array
DEBUG: begin partition::build_from_contiguous
DEBUG: end   partition::build_from_contiguous
DEBUG: begin partition::build_starting_indices
DEBUG: end   partition::build_starting_indices
DEBUG: begin copy
DEBUG: end   copy
DEBUG: begin copy
DEBUG: end   copy
DEBUG: begin dense::fill
DEBUG: end   dense::fill
DEBUG: begin dense::fill
DEBUG: end   dense::fill
DEBUG: begin dense::fill
DEBUG: end   dense::fill
DEBUG: begin repetition
DEBUG: begin dense::add_scaled
DEBUG: end   dense::add_scaled
DEBUG: end   repetition
DEBUG: end   axpy
	Running blas: scal
DEBUG: begin scal
DEBUG: begin partition::build_ranges_from_global_size
DEBUG: end   partition::build_ranges_from_global_size
DEBUG: begin components::fill_array
DEBUG: end   components::fill_array
DEBUG: begin components::fill_array
DEBUG: end   components::fill_array
DEBUG: begin components::fill_array
DEBUG: end   components::fill_array
DEBUG: begin components::fill_array
DEBUG: end   components::fill_array
DEBUG: begin partition::build_from_contiguous
DEBUG: end   partition::build_from_contiguous
DEBUG: begin partition::build_starting_indices
DEBUG: end   partition::build_starting_indices
DEBUG: begin copy
DEBUG: end   copy
DEBUG: begin copy
DEBUG: end   copy
DEBUG: begin dense::fill
DEBUG: end   dense::fill
DEBUG: begin dense::fill
DEBUG: end   dense::fill
DEBUG: begin repetition
DEBUG: begin dense::scale
DEBUG: end   dense::scale
DEBUG: end   repetition
DEBUG: end   scal
DEBUG: end   n = 100 
